pennylane/measurements.py

```diff
@@ -587,3 +593,115 @@ def circuit():
     """
     # pylint: disable=protected-access
     return MeasurementProcess(State, wires=qml.wires.Wires(wires))
+
+
+T = TypeVar("T")
+
+
+class MeasurementValueError(ValueError):
+    """Error raised when an unknown measurement value is being used."""
+
+
+class MeasurementValue(Generic[T]):
```

Comment:
Is `Generic[T]` useful later on?

Reply:
Yes, a `MeasurementValue` is capable of holding any type of object, although ya, useful for a next pr.